lib/logstash/filters/grok/timeout_enforcer.rb

```diff
@@ -27,7 +24,7 @@ def grok_till_timeout(event, grok, field, value)
       # It may appear that this should go in #stop_thread_groking but that would actually
       # break functionality! If this were moved there we would clear the interrupt
       # immediately after setting it in #cancel_timed_out, hence this MUST be here
-      thread.interrupted
+      java.lang.Thread.interrupted
```

Comment:
@andrewvc because it's a static method. From `java.lang.Thread`:

```java
    public static boolean interrupted() {
        return currentThread().isInterrupted(true);
    }
```

it's really confusing if we invoke it on an instance (and creates the false impression of clearing the interrupt on **some** `thread` when this clearly only works on the **current** thread).